### DISCUSSION TITLE
rowsep affected by other environment

### DISCUSSION THREAD

**Discussion by libinxu01** [lvjr/tabularray#383] [Q&A]
When I use ctex package, the abovesep and belowsep seems works different. I wonder there is a way to reset the longtblr environment.
Init:
<img width="569" alt="before" src="https://user-images.githubusercontent.com/15907992/225955207-486c4c8e-5a54-45d9-bbba-330f825bcb5a.png">
Use ctex package:
<img width="564" alt="after" src="https://user-images.githubusercontent.com/15907992/225955240-bad05016-204a-4821-9a00-3a6a8fd88a34.png">

```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xcolor}

\usepackage[
    UTF8,
    12pt,
    scheme=chinese,
    autoindent=true,
    heading=false,
    fontset=none,
    linespread=1.5]{ctex}

\begin{document}
	\begin{longtblr}[label=none]{
		colspec={XXX},
		rowspec={Q[15pt]Q[25pt]},
		row{1}={abovesep=5pt, belowsep=5pt}, 
		row{2}={abovesep=0pt,belowsep=0pt},
		vlines,hlines
	}
	communicate & eeeeeeeeeee & communication  \\
	communicate & eeeeeeeeeee & communication  \\
    \end{longtblr}
\end{document}
```

**Comment by muzimuzhi** [comment]
It's affected by `linespread` settings. Try
```tex
\AddToHook{env/longtblr/begin}{\linespread{1}\selectfont}
```

Similar discussion in zhihu.com, in Chinese https://www.zhihu.com/question/521756043

**Reply by libinxu01** [reply]
> It's affected by `linespread` settings. Try
> 
> ```
> \AddToHook{env/longtblr/begin}{\linespread{1}\selectfont}
> ```
> 
> Similar discussion in zhihu.com, in Chinese https://www.zhihu.com/question/521756043

It seems that rowsep is changed with linespread. How to use a fixed rowsep? I think the linespread should only affect text lines.

**Reply by muzimuzhi** [reply]
`rowsep/abovesep/belowsep` are irrelevant to `linespread`.  Currently, `linespread` affects height and depth of `\strutbox`, which affects the minimal row height and depth with the default `stretch` setting.

Did my previous suggestion not work for you?

**Reply by muzimuzhi** [reply]
> I think the linespread should only affect text lines.

Are rows in tables counted for text lines?  It depends.  Consecutive rows without h-lines look quite like text lines, and multi-line cell content is already text.  Not to mention the need to have consistent baseline skip and apply grid system to a document.

**Reply by libinxu01** [reply]
> `rowsep/abovesep/belowsep` are irrelevant to `linespread`. Currently, `linespread` affects height and depth of `\strutbox`, which affects the minimal row height and depth with the default `stretch` setting.
> 
> Did my previous suggestion not work for you?

I get it. The previous suggestion works. I want a fixed row height, rowspec sets the content height and rowsep sets the padding. The the rowsep is necessary because content might be crowd to hline without it. And in my cases the linespread can be set for different tables.

**Reply by muzimuzhi** [reply]
`stretch=0` is another option, see package doc (2023A, 2023-03-01), sec. 3.1.2 _Minimal Strut for Cell Text_, but then you need to adjust visual row separations by yourself.

**Reply by libinxu01** [reply]
I get it. Thank you so much!

**Comment by muzimuzhi** [comment]
> ```
> \usepackage[
>     UTF8,
>     12pt,
>     scheme=chinese,
>     autoindent=true,
>     heading=false,
>     fontset=none,
>     linespread=1.5]{ctex}
> ```

BTW, `\usepackage[linespread=1.5]{ctex}` sets the `\baselineskip` to 1.8 (1.2 * 1.5) times of current font size. See answers on zhihu.com [\[1\]][2], [\[2\]][2], and [`zhlineskip`](https://ctan.org/pkg/zhlineskip) package doc.

[1]: https://www.zhihu.com/question/62327906/answer/197899935
[2]: https://www.zhihu.com/question/58100147/answer/155635456

**Reply by libinxu01** [reply]
Get it ~
